corehq/messaging/scheduling/models/content.py

```diff
@@ -615,3 +615,91 @@ def get_recipient_devices_fcm_tokens(self, recipient):
         if not devices_fcm_tokens:
             raise FCMTokenValidationException(MessagingEvent.ERROR_NO_FCM_TOKENS)
         return devices_fcm_tokens
+
+
+def _meta_property(name):
+    def fget(self):
+        return getattr(self._meta, name)
+    return property(fget)
+
+
+class EmailImage(object):
+    """EmailImage is a thin wrapper around BlobMeta"""
+    id = _meta_property("id")
+    domain = _meta_property("parent_id")
+    filename = _meta_property("name")
+    blob_id = _meta_property("key")
+    content_type = _meta_property("content_type")
+    content_length = _meta_property("content_length")
+    delete_after = _meta_property("expires_on")
+
+    def __init__(self, meta):
+        self._meta = meta
+
+    @classmethod
+    def get(cls, domain, pk):
+        return cls(cls.meta_query(domain).get(pk=pk))
+
+    @classmethod
+    def get_by_key(cls, domain, key):
+        return cls(cls.meta_query(domain).get(key=key))
+
+    @classmethod
+    def get_by_keys(cls, keys, domain=None):
+        return [cls(m) for m in cls.meta_query(domain).filter(key__in=keys).all()]
+
+    @staticmethod
+    def meta_query(domain=None):
+        Q = models.Q
+        query = BlobMeta.objects.partitioned_query(domain).filter(
+            Q(expires_on__isnull=True) | Q(expires_on__gte=datetime.utcnow()),
+            type_code=CODES.email_multimedia,
+        )
+        if domain is not None:
+            query = query.filter(parent_id=domain)
+        return query
+
+    @classmethod
+    def get_all(cls, domain=None):
```

Comment:
Does this work in a partitioned DB if `domain` is not provided?  Ie does `.partitioned_query(None)` query on all partitions?

Reply:
This is a great point. I didn't have a partitioned DB on my local system when I was testing this.

Reply:
https://github.com/dimagi/commcare-hq/pull/33798/commits/114324ee75aad1d9825c22b645fbec87cc0f84ee